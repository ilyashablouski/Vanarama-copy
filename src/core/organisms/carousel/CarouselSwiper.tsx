import React, { memo } from 'react';
import cx from 'classnames';
import SwiperCore, { Navigation, Pagination } from 'swiper';
import { NavigationOptions, PaginationOptions } from 'swiper/types';
import { Swiper } from 'swiper/react';

import { ICarouselProps } from 'core/organisms/carousel/interface';

import ArrowBackSharp from 'core/assets/icons/ArrowBackSharp';
import ArrowForwardSharp from 'core/assets/icons/ArrowForwardSharp';

import Icon from 'core/atoms/icon';

SwiperCore.use([Navigation, Pagination]);

const navigationDefaultOptions: NavigationOptions = {
  prevEl: `.swiper-prev`,
  nextEl: `.swiper-next`,
};
const paginationDefaultOptions: PaginationOptions = {
  el: '.swiper-pagination',
  type: 'bullets',
  clickable: true,
};
const noSwipingSelector: string =
  'input, select, option, textarea, button, .button, video, label';

function CarouselSwiper({
  loop,
  countItems,
  watchOverflow,
  disableNavigation,
  className,
  children,
  onSlideChange,
  paginationComponent,
  navigationOptions = navigationDefaultOptions,
  paginationOptions = paginationDefaultOptions,
}: ICarouselProps) {
  return (
    <div className={cx('carousel', className)}>
      <Swiper
        slidesPerView="auto"
        loop={loop ?? countItems > 1}
        watchOverflow={watchOverflow}
        navigation={navigationOptions}
        pagination={paginationOptions}
        noSwipingSelector={noSwipingSelector}
        touchEventsTarget="wrapper"
        onSlideChange={onSlideChange}
      >
        {children}
        {paginationComponent || (
          <div
            className={cx('carousel--nav', {
              '-disabled': disableNavigation,
            })}
          >
            <button className="carousel--nav-arrow swiper-prev" type="button">
              <Icon icon={<ArrowBackSharp />} color="darker" size="regular" />
            </button>

            <div className="swiper-pagination" />

            <button className="carousel--nav-arrow swiper-next" type="button">
              <Icon
                icon={<ArrowForwardSharp />}
                color="darker"
                size="regular"
              />
            </button>
          </div>
        )}
      </Swiper>
    </div>
  );
}

export default memo(CarouselSwiper);
