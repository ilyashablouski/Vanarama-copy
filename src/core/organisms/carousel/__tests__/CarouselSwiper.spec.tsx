import React from 'react';
import { shallow } from 'enzyme';
import { SwiperSlide } from 'swiper/react';
import CarouselSwiper from '../CarouselSwiper';

describe('<CarouselSwiper />', () => {
  it('should render correctly with children', () => {
    const wrapper = shallow(
      <CarouselSwiper countItems={3}>
        <SwiperSlide>
          <div>
            <p>slide 1</p>
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div>
            <p>slide 2</p>
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div>
            <p>slide 3</p>
          </div>
        </SwiperSlide>
      </CarouselSwiper>,
    );

    // ASSERT
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly with two children', () => {
    const wrapper = shallow(
      <CarouselSwiper countItems={2}>
        <SwiperSlide>
          <div>
            <p>slide 1</p>
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div>
            <p>slide 2</p>
          </div>
        </SwiperSlide>
      </CarouselSwiper>,
    );

    // ASSERT
    expect(wrapper).toMatchSnapshot();
  });
});
