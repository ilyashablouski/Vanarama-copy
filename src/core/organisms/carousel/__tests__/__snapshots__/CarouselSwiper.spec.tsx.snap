// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselSwiper /> should render correctly with children 1`] = `
<div
  className="carousel"
>
  <Swiper
    autoHeight={true}
    loop={true}
    navigation={
      Object {
        "nextEl": ".swiper-next",
        "prevEl": ".swiper-prev",
      }
    }
    pagination={
      Object {
        "clickable": true,
        "el": ".swiper-pagination",
        "type": "bullets",
      }
    }
    slidesPerView={3}
    spaceBetween={20}
  >
    <SwiperSlide>
      <div>
        <p>
          slide 1
        </p>
      </div>
    </SwiperSlide>
    <SwiperSlide>
      <div>
        <p>
          slide 2
        </p>
      </div>
    </SwiperSlide>
    <SwiperSlide>
      <div>
        <p>
          slide 3
        </p>
      </div>
    </SwiperSlide>
  </Swiper>
  <div
    className="carousel--nav"
  >
    <button
      className="carousel--nav-arrow swiper-prev"
      type="button"
    >
      <Icon
        color="darker"
        icon={<Memo />}
        size="regular"
      />
    </button>
    <div
      className="swiper-pagination"
    />
    <button
      className="carousel--nav-arrow swiper-next"
      type="button"
    >
      <Icon
        color="darker"
        icon={<Memo />}
        size="regular"
      />
    </button>
  </div>
</div>
`;

exports[`<CarouselSwiper /> should render correctly with two children 1`] = `
<div
  className="carousel"
>
  <Swiper
    autoHeight={true}
    loop={true}
    navigation={
      Object {
        "nextEl": ".swiper-next",
        "prevEl": ".swiper-prev",
      }
    }
    pagination={
      Object {
        "clickable": true,
        "el": ".swiper-pagination",
        "type": "bullets",
      }
    }
    slidesPerView={2}
    spaceBetween={20}
  >
    <SwiperSlide>
      <div>
        <p>
          slide 1
        </p>
      </div>
    </SwiperSlide>
    <SwiperSlide>
      <div>
        <p>
          slide 2
        </p>
      </div>
    </SwiperSlide>
  </Swiper>
  <div
    className="carousel--nav"
  >
    <button
      className="carousel--nav-arrow swiper-prev"
      type="button"
    >
      <Icon
        color="darker"
        icon={<Memo />}
        size="regular"
      />
    </button>
    <div
      className="swiper-pagination"
    />
    <button
      className="carousel--nav-arrow swiper-next"
      type="button"
    >
      <Icon
        color="darker"
        icon={<Memo />}
        size="regular"
      />
    </button>
  </div>
</div>
`;
